File: base/test_db.js
const { createClient } = require('@supabase/supabase-js');

const supabaseUrl = "https://kikhexoxlkzofccnnkze.supabase.co";
const supabaseKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Imtpa2hleG94bGt6b2ZjY25ua3plIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzMwNzAwMzAsImV4cCI6MjA4ODY0NjAzMH0.wbBTASBQsP76PatkLCAt3Yv_A1IE_mgQ8pK17MB0uVk";
const supabase = createClient(supabaseUrl, supabaseKey);

async function runTest() {
    console.log("Fetching chat sessions...");
    const { data: sessions, error: sessionErr } = await supabase
        .from('chat_sessions')
        .select('*')
        .order('created_at', { ascending: false })
        .limit(2);

    if (sessionErr) console.error("Session Error:", sessionErr);
    else {
        console.log("Last 2 sessions:", JSON.stringify(sessions, null, 2));

        if (sessions && sessions.length > 0) {
            const sessionId = sessions[0].id;
            console.log(`\nFetching messages for session: ${sessionId}`);
            const { data: msgs, error: msgErr } = await supabase
                .from('chat_messages')
                .select('*')
                .eq('session_id', sessionId)
                .order('created_at', { ascending: true });

            if (msgErr) console.error("Msg Error:", msgErr);
            else console.log("Messages:", JSON.stringify(msgs, null, 2));
        }
    }
}

runTest();

File: app/test_db.py
import os
from supabase import create_client

supabase = create_client(
    os.environ.get("SUPABASE_URL", "https://kikhexoxlkzofccnnkze.supabase.co"),
    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Imtpa2hleG94bGt6b2ZjY25ua3plIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzMwNzAwMzAsImV4cCI6MjA4ODY0NjAzMH0.wbBTASBQsP76PatkLCAt3Yv_A1IE_mgQ8pK17MB0uVk"
)

res = supabase.table("transcriptions").select("id, word_count, text, created_at").order("created_at", desc=True).limit(1).execute()
if res.data:
    row = res.data[0]
    print(f"ID: {row['id']}")
    print(f"Words: {row['word_count']}")
    text = row['text']
    print(f"Last 500 chars:\n{text[-500:]}\n\nTotal length: {len(text)}")
else:
    print("No transcriptions found")
